### ISSUE TITLE
Opposite of F16C(i, m)

### ISSUE THREAD

**Issue by smohring** [PetteriAimonen/libfixmath#34]
Hi guys,

Iam wondering how i can get the "i" and "m" parts back out of a fixed point value?
Is there always a way ti achive that, oder do i have to do it manually (by extracting the single bits from the fixed point value)?

**Comment by PetteriAimonen** [comment]
I guess there are no functions / macros for it, but basically it is just `value >> 16` and `value & 0xFFFF`.

**Comment by smohring** [comment]
Hi @PetteriAimonen, thank you for your fast reply!

I was thinking not about the single bits, what i would really need is to get are the (modified) integers, that i used to construct a Fix16 instance with F16C(i,m).

Some functions like "setBeforeDot" / "setAfterDot" / "getBeforeDot" / "getAfterDot" with simple integers.

Another problem, iam facing right now, is, that F16C takes only constant values - i cannot use variables to build a Fix16 instance. I know, that i could do that with double - but i want to work completely without float types.

Do you think its possible, to extend that?

**Comment by PetteriAimonen** [comment]
Yeah, for convenience F16C macro accepts a decimal representation of the fractional part.
It is a string, not an integer.

But using decimal strings makes no sense for your own calculations, just use fractions (rational numbers).
